pyro/util.py

```diff
@@ -62,34 +64,66 @@ def torch_isinf(x):
     A convenient function to check if a Tensor contains any +inf; also works with numbers
     """
     if isinstance(x, numbers.Number):
-        return x == float('inf') or x == -float('inf')
-    return (x == float('inf')).any() or (x == -float('inf')).any()
+        return x == math.inf or x == -math.inf
+    return (x == math.inf).any() or (x == -math.inf).any()
 
 
-def warn_if_nan(value, msg=""):
+def warn_if_nan(value, msg="", *, filename=None, lineno=None):
     """
     A convenient function to warn if a Tensor or its grad contains any nan,
     also works with numbers.
     """
+    if filename is None:
+        try:
+            frame = sys._getframe(1)
+        except ValueError:
+            filename = "sys"
+            lineno = 1
+        else:
+            filename = frame.f_code.co_filename
+            lineno = frame.f_lineno
```

Comment:
Compare this logic to [warnings.py](https://github.com/python/cpython/blob/518835f3354d6672e61c9f52348c1e4a2533ea00/Lib/warnings.py#L298). Note `._getframe()` is private because it is a CPython detail, but PyTorch is limited to CPython.